x-pack/plugins/apm/public/plugin.ts

```diff
@@ -71,21 +66,47 @@ export class ApmPlugin implements Plugin<ApmPluginSetup, ApmPluginStart> {
     this.initializerContext = initializerContext;
   }
   public setup(core: CoreSetup, plugins: ApmPluginSetupDeps) {
-    createCallApmApi(core.http);
     const config = this.initializerContext.config.get();
     const pluginSetupDeps = plugins;
 
     pluginSetupDeps.home.environment.update({ apmUi: true });
     pluginSetupDeps.home.featureCatalogue.register(featureCatalogueEntry);
 
     if (plugins.observability) {
+      const getApmDataHelper = async () => {
+        const { fetchOverviewPageData, hasData } = await import(
+          './services/rest/apm_overview_fetchers'
+        );
+
+        return { fetchOverviewPageData, hasData };
+      };
       plugins.observability.dashboard.register({
         appName: 'apm',
-        fetchData: fetchOverviewPageData,
-        hasData,
+        hasData: async () => {
+          const dataHelper = await getApmDataHelper();
+          return await dataHelper.hasData();
```

Comment:
I think one of these changes somehow breaks the Observability overview page in the sense that it doesn't show any data. I haven't looked into why.

Reply:
@dgieselaar can you provide screenshot? for me it's working fine.

Reply:
![image](https://user-images.githubusercontent.com/352732/90406588-ade65300-e0a5-11ea-9988-fc8e356b25e1.png)

I also see this error in the console:

![image](https://user-images.githubusercontent.com/352732/90406616-b9d21500-e0a5-11ea-82cf-276f0ad34f32.png)

Reply:
I see the same error in the console as @dgieselaar but I don't see the UI issues